Fix the species filter dropdown

azangru opened the pull request:
## Bug
If you do a search on any eg site, you will see a dropdown filter that's supposed to restrict search results to a single species. The dropdown in production currently looks like this:

<img src="https://user-images.githubusercontent.com/6834224/91460578-dc231a00-e87f-11ea-9003-6712d3e2da55.png" width=600/>

## Causes
There are various problems related to this bug; and the old age of the code suggests that it has been present for a while:
- `modules/EnsEMBL/Web/Component/Search/Results.pm` is using the `DISPLAY_NAME` key, while the correct key, presumably, is `SPECIES_DISPLAY_NAME`
- After the `DISPLAY_NAME` was replaced with the `SPECIES_DISPLAY_NAME`, it turned out that the filter adds species display name to the url; and then in the `EBeyeSearch.pm` module tries to get species production name back from the display name.  For whatever reason, this didn't work; so I changed display name in the url to production name. Don't know if things are going to break because of that.

## Sandbox
http://ves-hx2-76.ebi.ac.uk:8450/Multi/Search/Results?q=PAD4;species=all;collection=all;site=ensemblthis

## Remaining Problems
- The `production_name` function still remains in the `EBeyeSearch.pm` module and is being used by the `ebeye_query` function. Don't know if it works at all.
- ~The `DISPLAY_NAME` key is still used in the `modules/EnsEMBL/Web/Document/HTML/HomeSearch.pm` module. I do not know whether this part of the code works either.~

azangru committed:
Fix the species filter dropdown

ens-ap5 commented:
Is there any need for that production_name method in EBeyeSearch if you've removed the calls to it? You seem to be trying to dereference a string on line 88.

azangru commented:
@ens-ap5 there's still the `ebeye_query` method that's [using it](https://github.com/EnsemblGenomes/eg-web-search/blob/release/eg/48/modules/EnsEMBL/Web/EBeyeSearch.pm#L103), and the string that it passes to the `production_name` method is not something coming from a query parameter

ens-ap5 commented:
Fair enough. I'm still a bit perplexed by the dereferencing, though - unless there's something I'm missing.

azangru commented:
The dropdown on the home page is also broken, by the way:

![image](https://user-images.githubusercontent.com/6834224/91544069-1b507a00-e917-11ea-80dc-97be12bf2546.png)


 This must be due to the use of the `DISPLAY_NAME` key in the HomeSearch.pm module. I'll change it to `SPECIES_DISPLAY_NAME`.

ens-st3 reviewed (commented):
How does this compare to one of the earlier archives (which I presume work!) ? Links to them are on the divisional home pages.

azangru commented:
@ens-st3 Archives do not seem to have EBEyeSearch plugin enabled.

But at least we know that the `DISPLAY_NAME` key works in archives.

See e.g. http://sep2019-plants.ensembl.org/index.html

azangru commented:
@ens-st3 the `Psychic` behaviour is also broken. Compare:

http://sep2019-plants.ensembl.org/Multi/Psychic?site=ensemblunit&species=Arabidopsis_thaliana&q=chx28 — Psychic will redirect to Arabidopsis_thaliana page

http://plants.ensembl.org/Multi/Psychic?site=ensemblunit&species=Arabidopsis_thaliana&q=chx28 — Psychic will replace `species=Arabidopsis_thaliana` with `species=all` and redirect to the search page for all species

ens-st3 reviewed (approved):
Functionally does what's needed

azangru commented:
@ens-ap5 , @ens-st3 the problem with Psychic is contained in this repository as well. Specifically, [here](https://github.com/EnsemblGenomes/eg-web-search/blob/release/eg/48/modules/EnsEMBL/Web/Controller/Psychic.pm#L213-L217):

```
    $species = 'all' if (lc($species) eq 'multi' or $site_type eq 'ensemblunit');
    if ($species and lc($species) ne 'all') {
      $species = $species_defs->get_config($species, 'SPECIES_COMMON_NAME');
      $species =~ s/(\[|\])//g;
    }
```

`$species_defs->get_config($species, 'SPECIES_COMMON_NAME');` returns `undef`, and Psychic redirects to the Multi search page.

So somehow, we should replicate the logic of "if it's a valid species, use the species page, but if it's an invalid species, redirect to the Multi search page".

Notice that the last line in the snippet above (`$species =~ s/(\[|\])//g;`) was added [here](https://github.com/EnsemblGenomes/eg-web-search/commit/465046e48c670476125019ca3288151240418cbc) ([ticket ENSEMBL-4264](https://www.ebi.ac.uk/panda/jira/browse/ENSEMBL-4264)) to deal with square brackets in some bacterial species names.

azangru committed:
Update EBeyeSearch and Psychic

Mapping of common name to production name doesn't seem to be working.
Here's a change with a hope that we don't even need such mapping.

azangru commented:
Ok, I got enough courage to remove the code that was expecting to receive display name and map it to production name, because this mapping isn't working anyway.

I also removed the redefining of the `$species` variable in Psychic (notice that the [example url](http://bacteria.ensembl.org/_bacillus_selenitireducens_mls10/Search/Results?species=Bacillus%20selenitireducens%20MLS10;idx=;q=Bsel_0282;site=ensemblthis) that was working in ENSEMBL-4264 four years ago is not working now anyway).

All changes seem to be working correctly in my Plants sandbox.
